client/webserver/site/src/js/markets.ts

```diff
@@ -1482,7 +1482,14 @@ export default class MarketsPage extends BasePage {
       details.side.classList.add(ord.sell ? 'sellcolor' : 'buycolor')
       header.side.classList.add(ord.sell ? 'sellcolor' : 'buycolor')
       details.qty.textContent = header.qty.textContent = Doc.formatCoinValue(ord.qty, market.baseUnitInfo)
-      details.rate.textContent = header.rate.textContent = Doc.formatRateFullPrecision(ord.rate, market.baseUnitInfo, market.quoteUnitInfo, cfg.ratestep)
+      let rate = ord.rate
+      let ratePrefix = ''
+      const nMatch = ord.matches?.length
+      if (ord.type === OrderUtil.Market && nMatch > 0) {
+        rate = OrderUtil.averageRate(ord)
+        if (nMatch > 1) ratePrefix = '~ ' // ~ only makes sense if the order has more than one match
+      }
+      details.rate.textContent = header.rate.textContent = ratePrefix + Doc.formatRateFullPrecision(rate, market.baseUnitInfo, market.quoteUnitInfo, cfg.ratestep)
```

Comment:
This will show `0.000000` if a market order is in epoch or was a no-match. That's pre-existing, but not ideal. Maybe this is where we should show the (translated) string `'market'` if we have no matches.